src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -25,11 +26,23 @@
 import com.github.kagkarlsson.scheduler.task.TaskInstance;
 
 public interface SchedulerClient {
-	
-	void scheduleForExecution(Instant exeecutionTime, TaskInstance taskInstance);
-	
+
+	/**
+	 * This method has been renamed and will be removed in a future version.
+	 *
+	 * @deprecated use {@link #schedule(Instant, TaskInstance)} instead.
+	 */
+	@Deprecated
+	void scheduleForExecution(Instant executionTime, TaskInstance taskInstance);
+
+	void schedule(Instant executionTime, TaskInstance taskInstance);
+
+	void reschedule(String taskName, String instanceId, Instant newExecutionTime);
```

Comment:
Nice! Doesn't look like it required too many changes to introduce the type. Might as well make it part of this PR.

Was thinking we should also change the order of parameters of the `schedule` method, for symmetry...  
```
void schedule(TaskInstance taskInstance, Instant executionTime);
```